presto-spi/src/main/java/io/prestosql/spi/connector/InMemoryRecordSet.java

```diff
@@ -243,6 +244,10 @@ else if (type instanceof RowType) {
                     checkArgument(value instanceof Block,
                             "Expected value %d to be an instance of Block, but is a %s", i, value.getClass().getSimpleName());
                 }
+                else if (type instanceof DecimalType) {
```

Comment:
Split this into two checks for `isShortDecimal` and `isLongDecimal`, then we can validate the type is correct for each one.